File: backend/src/models/UserStat.js
import { DataTypes } from 'sequelize';
import sequelize from '../../db.js';

const userStats = sequelize.define('userStats', {
    userId: { type: DataTypes.INTEGER, allowNull: false },
    totalFlights: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 0 },
    totalHours: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 0 },
    firstFlight: { type: DataTypes.DATE },
    lastFlight: { type: DataTypes.DATE },
    timesInFirstClass: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 0 },
    mostUsedAirline: { type: DataTypes.STRING(100) },
    totalSpent: { type: DataTypes.DECIMAL(10, 2), allowNull: false, defaultValue: 0 }
}, {
    tableName: 'userStats',
    timestamps: false
})
export default userStats

File: AetheriumBack/Models/UserStat.cs
namespace AetheriumBack.Models;

public class UserStat
{
    public int UserId { get; set; }
    public int TotalFlights { get; set; }
    public int TotalHours { get; set; }
    public DateTimeOffset FirstFlight { get; set; }
    public DateTimeOffset LastFlight { get; set; }
    public int TimesInFirstClass { get; set; }
    public string MostUsedAirline { get; set; }
    public Decimal TotalSpent { get; set; }

    private UserStat() { }

    public UserStat(string[] attributes)
    {
        //UserId = int.Parse(attributes[0]);
        TotalFlights = int.Parse(attributes[1]);
        TotalHours = int.Parse(attributes[2]);
        FirstFlight = DateTimeOffset.Parse(attributes[3]);
        LastFlight = DateTimeOffset.Parse(attributes[4]);
        TimesInFirstClass = int.Parse(attributes[5]);
        MostUsedAirline = attributes[6];
        TotalSpent = Decimal.Parse(attributes[7]);
    }
}
